bee-inx/src/node/config.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_block as bee;
+use inx::proto;
+
+use crate::field;
+
+/// The [`NodeConfiguration`] type.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct NodeConfiguration {
+    pub protocol_parameters: bee::protocol::ProtocolParameters,
+    pub milestone_public_key_count: u32,
+    // TODO: `milestone_key_ranges`
+    // TODO: `base_token`
+    pub supported_protocol_versions: Box<[u8]>,
+    // TODO: `pending_protocol_parameters`
```

Comment:
Are those to be done in this PR?

Reply:
Some of these, like `milestone_key_ranges` are still kind of blocked by Bee types, so I thought I'd do those later.

Reply:
I've implemented everything except for `milestone_key_ranges`.